### ISSUE TITLE
Integrate 1Password SDKs to support authentication via service account

### ISSUE THREAD

**Issue by carey404** [external-secrets/external-secrets#3655]
**Is your feature request related to a problem? Please describe.**
1Password recently released SDKs that support reading and writing secrets directly from 1Password without the need for a Connect server. This simplifies the integration process and reduces the dependency on additional infrastructure.

**Describe the solution you'd like**
Extend the 1Password integration to support the new 1Password SDKs. This would allow users to:

- Load secrets directly from 1Password without the need for hosting a Connect server.
- Retrieve, create, edit, and delete items in 1Password vaults.
- Authenticate using 1Password service accounts

**Describe alternatives you've considered**
Continuing to use the 1Password Connect server for integration, which requires additional setup and maintenance.

**Additional context**
The 1Password SDKs are currently in beta. Documentation can be found here: https://developer.1password.com/docs/sdks/

**Comment by mendymm** [comment]
@carey404 I am working on such an integration 
I will be honest, The main use case I have is reading in secrets not writing them to 1password
So in my head just reading in a simple op:// url is enough

But when you think of writing/editing/deleting a secret back to 1password, would you prefer a vault/item uuid or the op:// url?
The reason I ask is because the 1password go sdk only supports reading and writing items with vault and  item uuid

**Comment by carey404** [comment]
@mendymm Thanks for working on this! For now, focusing on reading secrets via the op:// URL should cover the majority of use cases.

If you were to implement writing, editing, and deleting items, I would suggest using the vault and item UUIDs since that’s what the 1Password SDK currently supports for these operations. It’s probably reasonable to wait and see if other users request this functionality before implementing it.

**Comment by mendymm** [comment]
> It’s probably reasonable to wait and see if other users request this functionality before implementing it.

I was thinking the same, Our use case is just to load secrets from 1pssword, not write/delete. For now I will continue on the work assuming the user will be mapping a op:// url to a single key in a secret

Let me know if you have any questions

**Comment by gusfcarvalho** [comment]
hey @mendymm ! Are you still working on this?

**Comment by mendymm** [comment]
> hey [@mendymm](https://github.com/mendymm) ! Are you still working on this?

No. Do to personal reasons, I won't be able to spend time on this.

**Comment by Skarlso** [comment]
I can finish the work.

**Comment by Skarlso** [comment]
@carey404 So I've been thinking about this. :) And I think a secret store should be /vault, right? 

So the format of the key is such as stated earlier:
```
op://<vault>/<item>/[section/]<field>
```

But this would mean that every external secret could access every vault always. That seems very dangerous.

Instead what I propose is that the secret store will have a `vault` field. And then further external secret's key needs to just use `item/section/field` format. Without the need to add `op://` at all.

How does that sound?

**Comment by Skarlso** [comment]
The PR is now finished and ready for review. I implemented the above because I feel like that's more inline with what the rest of the stores are doing.
